### PR TITLE
Python Azure: Add installation instructions for Azure in Python

### PR DESCRIPTION
**Pull request by lrafeei**
This PR adds instructions to install the New Relic Python agent when an application uses Microsoft Azure as a host

### CODE REVIEW

File: src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+def hello_world():
+    return "Hello, fellow humans!"
+
+if __name__ == '__main__':
+    app.run()
+```
+
+On the command line:
+
+```console
```
Comment (akristen) [inline src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx:47]:
```suggestion
```bash
```

or maybe something else? our decorator doesn't recognize "console"

Comment (lrafeei) [reply src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx:47]: Bash works!

File: src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+foo@bar:~$ cat requirements.txt 
+newrelic
+flask
+```
+
+If you're using a Dockerfile, add `newrelic-admin run-program` as an entrypoint to your startup script.
+
+```lang-dockerfile
```
[the remark this answers is not in the document]
Comment (lrafeei) [reply src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx:55]:
```bash can work here just fine for our purposes

Dockerfile is the filetype here.  Sorry about the confusion!  It seems like different markdown interpreters have different supports.

[the remark this answers is not in the document]
Comment (akristen) [reply src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx:55]: Yes! I'm not sure what our docs site uses, but afaik it supports frameworks and only some of them.

File: src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+* A workng installation of Python
+* The Windows Azure SDK for Python for your platform
+
+<Callout variant="important">
+  The Python agent cannot capture telemetry for Azure Functions. Instead, you will need to use the [Azure Functions monitoring integration](/docs/infrastructure/microsoft-azure-integrations/azure-integrations-list/azure-functions-monitoring-integration/).
+</Callout>
+
+## Installing the Python agent [#install_python]
+
+To add the New Relic Python agent to your Azure application, add `newrelic` to your `requirements.txt` file.
```
Comment (akristen) [inline src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx:25]: What do you think about pointing customers to the Python install doc, then prompting them to return to this doc? Are these separate install instructions not present in the Python install doc? @lrafeei

File: src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+Here's an example with a sample app:
```
Comment (akristen) [inline src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx:27]: Azure provides certain quickstarts in their docs for both managed services:

* [Container Apps](https://learn.microsoft.com/en-us/azure/container-apps/quickstart-portal)
* [App Service](https://learn.microsoft.com/en-us/azure/app-service/quickstart-python?tabs=flask%2Cwindows%2Cazure-cli%2Cazure-cli-deploy%2Cdeploy-instructions-azportal%2Cterminal-bash%2Cdeploy-instructions-zip-azcli)

I wonder if providing this sample code might risk it falling "out of sync" with Azure's docs. What if we were to direct customers to these quickstarts in the pre-reqs, replace this step altogether with something general (e.g., "Add the following code snippet to the [specific-file-whathaveyou] in your app", or to one of the Azure quickstarts.")

What do you think?

File: src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    app.run()
+```
+
+## Adding app settings in Azure [#azure_settings]
+
+After installation, you must configure your app. 
+
+### Azure App Services [#app-service-configuration-portal]
+
+1. Sign in to <DNT>**[portal.azure.com](https://portal.azure.com)**</DNT>.
```
Comment (akristen) [inline src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx:94]: Rather than providing steps about how a third party UI exists today, we describe that they need to add the variables to their App settings, then link out to a doc about App settings in the Azure docs? I think about how often the New Relic platform UI changes and how quickly docs can fall out of sync when trying to document a product whose development we have no insight into.

File: src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx (hunk 4 above)
Comment (lrafeei) [reply src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx:27]: My only concern with this is that those quickstarts, while convenient, utilize a lot of Behind-The-Scenes functionality within Azure.  Also, and this is my personal bias, the Azure docs are not very thorough and lack transparency.  In attempting to make the tutorials easy to follow, they often neglect to point out how to navigate common use cases for customers.  My intention here is to use a barebones example that makes the what and where of the changes needed evident.

Comment (akristen) [reply src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx]: Awesome, I'll stick with what you have!!

File: src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx (hunk 5 above)
Comment (akristen) [reply src/content/docs/apm/agents/python-agent/hosting-services/python-agent-microsoft-azure.mdx]: Let me think about this a bit tbh... It might be easier to keep what you have for now and if a problem comes up, we can change course. I'm not as familiar with Azure's docs or their doc hygiene -- as of right now, we don't have a way to detect 404'd links so we'd be relying on their diligence about embedding redirects with doc changes

File: src/content/docs/apm/agents/python-agent/hosting-services/python-azure-containerapps-appservice.mdx [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+CMD ["flask", "run"]
+```
+
+Otherwise, initialize the agent by adding these lines to your app:
+
+```python
+# app.py
+# Run with command "flask run"
+
+import newrelic.agent   # ADDED
```
Comment (akristen) [inline src/content/docs/apm/agents/python-agent/hosting-services/python-azure-containerapps-appservice.mdx:72]: Is initializing the same as adding the Python agent?

Comment (lrafeei) [reply src/content/docs/apm/agents/python-agent/hosting-services/python-azure-containerapps-appservice.mdx:72]: Sort of--the way the agent works is that you install the agent on the host (in this case, Azure), but you have to do something to let the app know that it needs to run the code with New Relic.  This is done either with the New Relic commands in front of the startup script or the initialization function.

TL;DR: I wanted to distinguish between installing New Relic and actually flagging New Relic to run on the app

Comment (akristen) [reply src/content/docs/apm/agents/python-agent/hosting-services/python-azure-containerapps-appservice.mdx]: Ah ok, and you install New Relic Python agent to the Azure app by running a bash command, "adding" New Relic to the flask app

File: src/content/docs/apm/agents/python-agent/hosting-services/python-azure-containerapps-appservice.mdx [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+        return "Hello, fellow humans!"
+
+    if __name__ == '__main__':
+        app.run()
+    ```
+    </Step>
+    <Step>
+        ## Configure your Azure settings [#azure-settings]
+
+Once you've installed the Python agent, we recommend configuring your app in either the Azure App Service or Container Apps portal, depending on the service you're using. You can update your app settings through the Azure UI, or using the Python SDK CLI.
```
Comment (lrafeei) [inline src/content/docs/apm/agents/python-agent/hosting-services/python-azure-containerapps-appservice.mdx:98]: In this case, it's just the "Python CLI" instead of the "Python SDK CLI"

File: src/content/docs/apm/agents/python-agent/hosting-services/python-azure-containerapps-appservice.mdx [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+    </Step>
+    <Step>
+        ## Configure your Azure settings [#azure-settings]
+
+Once you've installed the Python agent, we recommend configuring your app in either the Azure App Service or Container Apps portal, depending on the service you're using. You can update your app settings through the Azure UI, or using the Python SDK CLI. 
+
+<CollapserGroup>
+    <Collapser
+        id="config-app-services-ui"
+        title="Configure your app in Azure App Services"
```
Comment (lrafeei) [inline src/content/docs/apm/agents/python-agent/hosting-services/python-azure-containerapps-appservice.mdx:103]:
```suggestion
        title="Configure your app in Azure App Services through the Azure Portal"
```

File: src/content/docs/apm/agents/python-agent/hosting-services/python-azure-containerapps-appservice.mdx [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+3. Add the following to the <DNT>**App settings**</DNT>:
+    * `NEW_RELIC_APP_NAME`: Your Linux Azure website name
+    * `NEW_RELIC_LICENSE_KEY`: Your New Relic <InlinePopover type="licenseKey" />
+4. Save your settings.
+5. Restart your Python app.
+
+    </Collapser>
+    <Collapser
+        id="config-container-apps-ui"
+        title="Configure your app in Azure Container Apps"
```
Comment (lrafeei) [inline src/content/docs/apm/agents/python-agent/hosting-services/python-azure-containerapps-appservice.mdx:117]:
```suggestion
        title="Configure your app in Azure Container Apps though the Azure Portal"
```

File: src/content/docs/apm/agents/python-agent/hosting-services/python-azure-containerapps-appservice.mdx [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+   * `NEW_RELIC_APP_NAME`: Your Linux Azure website name
+   * `NEW_RELIC_LICENSE_KEY`: Your New Relic <InlinePopover type="licenseKey" />
+4. Save your settings.
+5. Restart your Python app.
+
+Generate some traffic to your app and wait a few minutes. You can see your data by going to <DNT>**[one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > APM & services > (select your app)**</DNT>.
+    </Collapser>
+    <Collapser
+    id="config-app-services-cli"
+    title="Configure your app in Azure Container Apps"
```
Comment (lrafeei) [inline src/content/docs/apm/agents/python-agent/hosting-services/python-azure-containerapps-appservice.mdx:132]:
```suggestion
    title="Configure your app in Azure App Services through the CLI"
```

File: src/content/docs/apm/agents/python-agent/hosting-services/python-azure-containerapps-appservice.mdx [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+Run the following on the command line: 
+
+```bash
+az webapp config appsettings set --name ${WEB_APP_NAME} --settings NEW_RELIC_LICENSE_KEY=${NEW_RELIC_LICENSE_KEY} NEW_RELIC_APP_NAME=${NEW_RELIC_APP_NAME}
+```
+
+    </Collapser>
+    <Collapser
+    id="config-container-apps-cli"
+    title="Configure your app in Azure Container Apps"
```
Comment (lrafeei) [inline src/content/docs/apm/agents/python-agent/hosting-services/python-azure-containerapps-appservice.mdx:144]:
```suggestion
    title="Configure your app in Azure Container Apps though the CLI"
```

File: src/content/docs/apm/agents/python-agent/hosting-services/python-azure-containerapps-appservice.mdx (hunk 7 above)
Comment (akristen) [inline src/content/docs/apm/agents/python-agent/hosting-services/python-azure-containerapps-appservice.mdx:98]:
```suggestion
Once you've installed the Python agent, we recommend configuring your app in either the Azure App Service or Container Apps portal, depending on the service you're using. You can update your app settings through the Azure UI, or using the Python CLI. 
```

**Review by akristen — APPROVED** [review]
LET'S GO! :)
